File: app/src/main/java/org/ametro/model/entities/MapRect.java
package org.ametro.model.entities;

public class MapRect {
    public final int x;
    public final int y;
    public final int width;
    public final int height;

    public MapRect(int x, int y, int width, int height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }

    public boolean contains(MapPoint point) {
        return point.x >= x && point.x <= (x + width) && point.y >= y && point.y <= (y + height);
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/model/entities/MapRect.kt
package io.github.romangolovanov.apps.ametro.model.entities

class MapRect(
    val x: Int,
    val y: Int,
    val width: Int,
    val height: Int
) {
    fun contains(point: MapPoint): Boolean {
        return point.x >= x && point.x <= (x + width) && point.y >= y && point.y <= (y + height)
    }
}
